L1Trigger/L1THGCal/src/be_algorithms/HGCalClusteringImpl.cc

```diff
@@ -270,3 +276,161 @@ void HGCalClusteringImpl::clusterizeNN( const edm::PtrVector<l1t::HGCalTriggerCe
 
 }
 
+
+
+/*** FW-algorithms ***/
+void HGCalClusteringImpl::clusterizeDRNN( const std::vector<edm::Ptr<l1t::HGCalTriggerCell>> & triggerCellsPtrs, 
+                                        l1t::HGCalClusterBxCollection & clusters,
+                                        const HGCalTriggerGeometryBase & triggerGeometry
+    ){
+
+    bool isSeed[triggerCellsPtrs.size()];
+    std::vector<unsigned> seedPositions;
+    seedPositions.reserve( triggerCellsPtrs.size() );
+
+    /* search for cluster seeds */
+    int itc(0);
+    for( std::vector<edm::Ptr<l1t::HGCalTriggerCell>>::const_iterator tc = triggerCellsPtrs.begin(); tc != triggerCellsPtrs.end(); ++tc,++itc ){
+        
+        double seedThreshold = ((*tc)->subdetId()==HGCHEB ? scintillatorSeedThreshold_ : siliconSeedThreshold_);
+
+        /* decide if is a seed, if yes store the position into of triggerCellsPtrs */
+        isSeed[itc] = ( (*tc)->mipPt() > seedThreshold) ? true : false;
+        if( isSeed[itc] ) {
+
+            seedPositions.push_back( itc );
+
+            /* remove tc from the seed vector if is a NN of an other seed*/
+            for( auto pos : seedPositions ){
+                if( ( (*tc)->position() - triggerCellsPtrs[pos]->position() ).mag()  < dr_ ){
+                    if( this->areTCneighbour( (*tc)->detId(), triggerCellsPtrs[pos]->detId(), triggerGeometry ) )
+                    {
+                        isSeed[itc] = false;
+                        seedPositions.pop_back();
+                    }
+                }
+            } 
+        }
+
+    }
+    
+    /* clustering the TCs */
+    std::vector<l1t::HGCalCluster> clustersTmp;
+
+    // every seed generates a cluster
+    for( auto pos : seedPositions ) {
+        clustersTmp.emplace_back( triggerCellsPtrs[pos] );
```

Comment:
This actually adds the seed in the constituents of the cluster.
The loop over the TC afterwards results then in a double counting for the seeds.
Working on a fix

Reply:
Fix available in #166